geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/SetClusterConfigurationCommand.java

```diff
@@ -0,0 +1,236 @@
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.xml.sax.SAXException;
+
+import org.apache.geode.cache.execute.ResultCollector;
+import org.apache.geode.distributed.ConfigurationPersistenceService;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.distributed.internal.InternalConfigurationPersistenceService;
+import org.apache.geode.management.cli.CliMetaData;
+import org.apache.geode.management.internal.cli.AbstractCliAroundInterceptor;
+import org.apache.geode.management.internal.cli.GfshParseResult;
+import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
+import org.apache.geode.management.internal.cli.i18n.CliStrings;
+import org.apache.geode.management.internal.cli.remote.CommandExecutionContext;
+import org.apache.geode.management.internal.cli.result.FileResult;
+import org.apache.geode.management.internal.cli.result.model.InfoResultModel;
+import org.apache.geode.management.internal.cli.result.model.ResultModel;
+import org.apache.geode.management.internal.cli.result.model.TabularResultModel;
+import org.apache.geode.management.internal.configuration.domain.Configuration;
+import org.apache.geode.management.internal.configuration.functions.GetRegionNamesFunction;
+import org.apache.geode.management.internal.configuration.functions.RecreateCacheFunction;
+import org.apache.geode.management.internal.security.ResourceOperation;
+import org.apache.geode.security.ResourcePermission;
+
+
+public class SetClusterConfigurationCommand extends InternalGfshCommand {
+  public static final String COMMAND = "set-cluster-configuration";
+  public static final String COMMAND_HELP =
+      "this commands overwrites the existing cluster configuration for the specified group.";
+  public static final String XML_FILE = "xml-file";
+  public static final String CONFIGURE_RUNNING_SERVERS = "configure-running-servers";
+  public static final String CONFIGURE_HELP =
+      "Running servers will be configured with the content of xml file only when the these servers do not have any existing configurations.";
+  public static final String IGNORE_RUNNING_SERVERS = "ignore-running-servers";
+  public static final String IGNORE_HELP =
+      "Running servers will be left alone. It's assumed that the content of the xml you are setting here has no conflict with these servers' configuration.";
+
+
+  @CliCommand(value = COMMAND, help = COMMAND_HELP)
+  @CliMetaData(
+      interceptor = "org.apache.geode.management.internal.cli.commands.SetClusterConfigurationCommand$Interceptor",
+      relatedTopic = {CliStrings.TOPIC_GEODE_CONFIG})
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
```

Comment:
We should probably require complete permissions for this command, shouldn't we?  While most of the gfsh-equivalent operations would be satisfied by `CLUSTER:MANAGE`, there are some that aren't.  The creation of a persistent queue comes to mind, which requires `CLUSTER:WRITE:DISK`.

I can't think off hand of anything that might require a `DATA` permission that this could enact, but I think maybe `CLUSTER:*` would be a better requirement here.

Reply:
it should require the same privilege as "export cluster-config" and "import cluster-config".